eth/handler.go

```diff
@@ -703,8 +703,9 @@ func (pm *ProtocolManager) handleMsg(p *peer) error {
 
 		// Mark the peer as owning the block and schedule it for import
 		p.MarkBlock(request.Block.Hash())
-		pm.fetcher.Enqueue(p.id, request.Block)
-
+		if !pm.blockchain.HasBlock(request.Block.Hash(), request.Block.NumberU64()) {
+			pm.fetcher.Enqueue(p.id, request.Block)
+		}
```

Comment:
I am not entirely sure this is the correct place to put it. Perhaps Enqueue should be extended to do this check internally, before propagating it to the inner loop (same I suggested for the txFetcher PR). That way we could even meaningfully meter it (vs. here we can't track the event, at least not in the correct scope).

Reply:
I pushed a change now, so it's more like you described it, and also added metrics so we have both the total sum of announces/broadcasts, as well as a metric of the 'useful' ones